scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/termination/handlers/TerminateReplicateTaskHandler.java

```diff
@@ -0,0 +1,240 @@
+package org.ow2.proactive.scheduler.job.termination.handlers;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.log4j.Logger;
+import org.ow2.proactive.scheduler.common.NotificationData;
+import org.ow2.proactive.scheduler.common.SchedulerEvent;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.common.task.TaskStatus;
+import org.ow2.proactive.scheduler.common.task.flow.FlowAction;
+import org.ow2.proactive.scheduler.common.task.flow.FlowBlock;
+import org.ow2.proactive.scheduler.core.SchedulerStateUpdate;
+import org.ow2.proactive.scheduler.job.ChangedTasksInfo;
+import org.ow2.proactive.scheduler.job.InternalJob;
+import org.ow2.proactive.scheduler.job.JobInfoImpl;
+import org.ow2.proactive.scheduler.task.internal.InternalTask;
+
+import it.sauronsoftware.cron4j.Predictor;
+
+public class TerminateReplicateTaskHandler {
+
+	public static final Logger logger = Logger.getLogger(TerminateReplicateTaskHandler.class);
+
+	private final InternalJob internalJob;
+
+	public TerminateReplicateTaskHandler(InternalJob internalJob) {
+		this.internalJob = internalJob;
+	}
+
+	public boolean terminateReplicateTask(FlowAction action, InternalTask initiator, ChangedTasksInfo changesInfo,
+			SchedulerStateUpdate frontend, TaskId taskId) {
+		int runs = action.getDupNumber();
+		if (runs < 1) {
+			runs = 1;
+		}
+
+		logger.info("Control Flow Action REPLICATE (runs:" + runs + ")");
+		List<InternalTask> toReplicate = new ArrayList<>();
+
+		// find the tasks that need to be replicated
+		for (InternalTask ti : internalJob.getIHMTasks().values()) {
+			List<InternalTask> tl = ti.getIDependences();
+			if (tl != null) {
+				for (InternalTask ts : tl) {
+					if (ts.getId().equals(initiator.getId()) && !toReplicate.contains(ti)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)